### PR TITLE
Implemented Reverse polish notation and tests.

### LINKED ISSUE #4: Reverse polish notation

### PR DESCRIPTION
**Pull request by admiralWoop**
Closes #4

### CODE REVIEW

**Comment by Rizzen** [comment]
@Onikiro Could you take a look?

**Comment by Rizzen** [comment]
@admiralWoop create subfolder in Algorithms project and place your implementation there

**Comment by admiralWoop** [comment]
@Rizzen will be done in a few minutes

**Comment by admiralWoop** [comment]
@Rizzen should I also create a new project for my implementation?

**Comment by Rizzen** [comment]
> 
> 
> @Rizzen should I also create a new project for my implementation?

No, it isn't necessary

**Comment by admiralWoop** [comment]
@Rizzen moved implementations and tests to individual folders

File: Algorithms/Algorithms/ReversePolishNotation.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+using System.Text.RegularExpressions;
+
+namespace Algorithms
+{
+    public class ReversePolishNotation
+    {
+        public static List<string> Parse(string expression)
+        {
+            List<string> parsed = new List<string>();
+            foreach (var a in Regex.Split(expression, @"((?<!\d)-?\d+\.?\d*)|([\+\-\*\/\^])|(\()|(\))"))
```
Comment (Onikiro) [inline Algorithms/Algorithms/ReversePolishNotation.cs]: Rename  'a' to more readable

File: Algorithms/Algorithms/ReversePolishNotation.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Algorithms
+{
+    public class ReversePolishNotation
+    {
+        public static List<string> Parse(string expression)
+        {
+            List<string> parsed = new List<string>();
```
Comment (Onikiro) [inline Algorithms/Algorithms/ReversePolishNotation.cs]: "parsed" - parsed tokens? Can you rename that too.

File: Algorithms/Algorithms/ReversePolishNotation.cs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                if (operators.Peek() == '(' || operators.Peek() == ')')
+                    throw new ArithmeticException("Brackets isn't balanced!");
+                output.Enqueue(operators.Pop().ToString());
+            }
+            return output.ToList();
+        }
+
+        public static double Calculate(List<string> expression)
+        {
+            Stack<double> expressions = new Stack<double>();
```
Comment (Onikiro) [inline Algorithms/Algorithms/ReversePolishNotation.cs]: var

File: Algorithms/Algorithms/ReversePolishNotation.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        public static List<string> Parse(string expression)
+        {
+            List<string> parsed = new List<string>();
+            foreach (var a in Regex.Split(expression, @"((?<!\d)-?\d+\.?\d*)|([\+\-\*\/\^])|(\()|(\))"))
+            {
+                if (!String.IsNullOrWhiteSpace(a))
+                    parsed.Add(a);
+            }
+
+            Queue<string> output = new Queue<string>();
```
Comment (Onikiro) [inline Algorithms/Algorithms/ReversePolishNotation.cs]: var

File: Algorithms/Algorithms/ReversePolishNotation.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            Stack<char> operators = new Stack<char>();
```
Comment (Onikiro) [inline Algorithms/Algorithms/ReversePolishNotation.cs]: var

File: Algorithms/Algorithms/ReversePolishNotation.cs [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        public static double Calculate(List<string> expression)
+        {
+            Stack<double> expressions = new Stack<double>();
+            var switchDict = new Dictionary<string, Action>()
+            {
+                { "^" , () => { var b = expressions.Pop();
+                                expressions.Push(Math.Pow(expressions.Pop(), b));} },
+                { "*" , () => expressions.Push(expressions.Pop() * expressions.Pop())},
+                { "/" , () => { var a = expressions.Pop();
+                                expressions.Push(expressions.Pop() / a); } },
```
Comment (Onikiro) [inline Algorithms/Algorithms/ReversePolishNotation.cs]: May be more spacing for simplification code reading?

File: Algorithms/Algorithms.Tests/ReversePolishNotationTests.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+using System;
+using Xunit;
+
+namespace Algorithms.Tests
+{
+    public class ReversePolishNotationTests
+    {
+        [Theory]
+        [InlineData("1+1", 2)]
```
Comment (Onikiro) [inline Algorithms/Algorithms.Tests/ReversePolishNotationTests.cs]: May need more examples with spaces and without.
For example:
2+2*2 and 2 +2* 2
can that example pass tests?
And for every token.

Comment (Onikiro) [reply Algorithms/Algorithms.Tests/ReversePolishNotationTests.cs]: And check negative numbers

File: Algorithms/Algorithms/ReversePolishNotation.cs (hunk 2 above)
Comment (Onikiro) [reply Algorithms/Algorithms/ReversePolishNotation.cs]: and var

**Comment by admiralWoop** [comment]
As you requested I changed explicit collections declarations to implicit.
Also i added RPN algorithm description to master README file.

**Comment by admiralWoop** [comment]
@Onikiro @Rizzen

**Comment by Onikiro** [comment]
Thanks for PR! 
Your changes was merged into repository.

**Pull request merged**